homeassistant/components/arcam_fmj/config_flow.py

```diff
@@ -1,12 +1,81 @@
 """Config flow to configure the Arcam FMJ component."""
-from operator import itemgetter
+import asyncio
+import logging
+from urllib.parse import urlparse
+
+import aiohttp
+from defusedxml import ElementTree
+import voluptuous as vol
 
 from homeassistant import config_entries
+from homeassistant.components.ssdp import ATTR_SSDP_LOCATION, ATTR_UPNP_UDN
 from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.data_entry_flow import AbortFlow
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    CONF_UUID,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DOMAIN,
+    DOMAIN_DATA_CONFIG,
+    DOMAIN_DATA_ENTRIES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _strip_uuid_prefix(data):
+    assert data.startswith("uuid:"), "Missing uuid prefix"
+    return data[5:].split("-")[4]
+
+
+async def _get_uniqueid_from_device_description(hass, url):
+    """Retrieve and extract unique id from url."""
+    try:
+        session = async_get_clientsession(hass)
+        async with session.get(url) as req:
+            req.raise_for_status()
+            xml = ElementTree.fromstring(await req.text())
+
+        namespaces = {"d": "urn:schemas-upnp-org:device-1-0"}
+        udn = xml.findtext("d:device/d:UDN", None, namespaces)
+        assert udn, "Failed to get udn tag in xml"
```

Comment:
assert checks can be disabled, should not be used. 

Also please break up the try block into wrapping and catching just the appropriate exceptions.